sstables/sstable_directory.cc

```diff
@@ -165,10 +165,12 @@ sstable_directory::highest_version_seen() const {
     return _max_version_seen;
 }
 
-future<>
-sstable_directory::process_sstable_dir(process_flags flags) {
+future<> sstable_directory::process_sstable_dir(process_flags flags) {
     dirlog.debug("Start processing directory {} for SSTables", _sstable_dir);
+    return _lister->process(flags, _sstable_dir, *this);
+}
 
+future<> sstable_directory::components_lister::process(process_flags flags, fs::path location, sstable_directory& directory) {
```

Comment:
nit: I'd reverse the order of parameters to be top-down: directory, location, flags
since the operand for `process()` is the location in directory, and flags additionally modify the function's behavior

Reply:
Agreed. Will do